backend/tests/unit/dal/Services/Admin/ProjectServiceTest.cs

```diff
@@ -0,0 +1,299 @@
+                new object[] { new ProjectFilter() { StatusId = new int[] { 2 } }, 1 }
+            };
+
+        public static IEnumerable<object[]> Workflows =>
+            new List<object[]>
+            {
+                new object[] { "SubmitDisposal", 6 },
+                new object[] { "ReviewDisposal", 1 }
+            };
+        #endregion
+
+        #region Constructors
+        public ProjectServiceTest() { }
+        #endregion
+
+        #region Tests
+        #region Get
+        [Fact]
+        public void Get_ByProjectNumber()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.ProjectView, Permissions.AdminProjects, Permissions.AgencyAdmin);
+            var project = EntityHelper.CreateProject(1);
+            helper.CreatePimsContext(user).AddAndSaveChanges(project);
+
+            var service = helper.CreateService<ProjectService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            var result = service.Get($"SPP-{1:00000}");
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(project, result, new ShallowPropertyCompare());
+        }
+
+        [Fact]
+        public void Get_ById()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.ProjectView, Permissions.AdminProjects, Permissions.AgencyAdmin);
+            var project = EntityHelper.CreateProject(1);
+            helper.CreatePimsContext(user).AddAndSaveChanges(project);
+
+            var service = helper.CreateService<ProjectService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            var result = service.Get(1);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(project, result, new ShallowPropertyCompare());
+        }
+
+        [Fact]
+        public void Get_Snapshots()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.ProjectView, Permissions.AdminProjects, Permissions.AgencyAdmin);
+            var project = EntityHelper.CreateProject(1);
+            var snap = EntityHelper.CreateProjectSnapshot(1, agency: project.Agency);
+            snap.ProjectId = project.Id;
+            var expectedCount = 1;
+            helper.CreatePimsContext(user).AddAndSaveChanges(snap);
+
+            var service = helper.CreateService<ProjectService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            var result = service.GetSnapshots(1);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(expectedCount, result.Count());
+        }
+
+                [Fact]
```

Comment:
spacing?

Reply:
Yeah trying to get my formatter to behave lol, I'll fix before I merge